### PR TITLE
Громов Алексей.  Задача 1. Вариант 2. Технология SEQ. Умножение плотных матриц. Элементы типа double. Блочная схема, алгоритм Фокса.

### PR DESCRIPTION
**Pull request by DSFKnight**
Код реализует последовательное умножение двух плотных матриц **A** и **B** типа `double` с использованием блочной схемы и алгоритма Фокса.

`PreProcessingImpl()`:
- Делит входные данные на две матрицы `A_` и `B_`.
- Определяет размер матрицы `n_` и размер блока `block_size_`.
- Проверяет, что матрица квадратная.

`ValidationImpl()`
- Проверяет корректность входных данных: размер должен быть четным (две матрицы), а сами матрицы — квадратными.

`RunImpl()`
- Делит матрицы на блоки размером `block_size_`.
- Выполняет блочное умножение в несколько этапов (стадий).
- Использует три вложенных цикла для вычисления произведения блоков.

`PostProcessingImpl()`
- Копирует результат из `output_` в выходной буфер.

### CODE REVIEW

File: tasks/seq/gromov_a_fox_algorithm/func_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  ASSERT_TRUE(test_task_sequential.Validation());
+  test_task_sequential.PreProcessing();
+  test_task_sequential.Run();
+  test_task_sequential.PostProcessing();
+
+  for (size_t i = 0; i < out.size(); ++i) {
+    EXPECT_NEAR(out[i], expected[i], 1e-9);
+  }
+}
+
```
Comment (ofuuse) [inline tasks/seq/gromov_a_fox_algorithm/func_tests/main.cpp:115]: Add a test with negative values

File: tasks/seq/gromov_a_fox_algorithm/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  ASSERT_TRUE(test_task_sequential.Validation());
+  test_task_sequential.PreProcessing();
+  test_task_sequential.Run();
+  test_task_sequential.PostProcessing();
+
+  std::vector<double> expected(kN * kN, static_cast<double>(kN));
+  for (size_t i = 0; i < out.size(); ++i) {
+    EXPECT_NEAR(out[i], expected[i], 1e-9);
+  }
+}
```
Comment (ofuuse) [inline tasks/seq/gromov_a_fox_algorithm/func_tests/main.cpp:143]: add more tests

**Review by ofuuse — CHANGES_REQUESTED** [review]

File: tasks/seq/gromov_a_fox_algorithm/func_tests/main.cpp [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  std::vector<double> a = {1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0};
+  std::vector<double> b = {16.0, 15.0, 14.0, 13.0, 12.0, 11.0, 10.0, 9.0, 8.0, 7.0, 6.0, 5.0, 4.0, 3.0, 2.0, 1.0};
+  std::vector<double> out(kN * kN, 0.0);
+
+  std::vector<double> expected = {80.0,  70.0,  60.0,  50.0,  240.0, 214.0, 188.0, 162.0,
+                                  400.0, 358.0, 316.0, 274.0, 560.0, 502.0, 444.0, 386.0};
+
+  std::vector<double> input;
+  input.insert(input.end(), a.begin(), a.end());
+  input.insert(input.end(), b.begin(), b.end());
```
Comment (WholeHorse) [inline tasks/seq/gromov_a_fox_algorithm/func_tests/main.cpp:23]: i think it will be better if you used std::copy instead insert in all tests

File: tasks/seq/gromov_a_fox_algorithm/src/ops_seq.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            }
+          }
+        }
+      }
+    }
+  }
+  return true;
+}
+
+bool gromov_a_fox_algorithm_seq::TestTaskSequential::PostProcessingImpl() {
```
Comment (WholeHorse) [inline tasks/seq/gromov_a_fox_algorithm/src/ops_seq.cpp:60]: here you can also use std::copy

**Review by WholeHorse — CHANGES_REQUESTED** [review]

**Review by ofuuse — APPROVED** [review]

**Review by WholeHorse — APPROVED** [review]

File: tasks/seq/gromov_a_fox_algorithm/func_tests/main.cpp [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <iterator>
+#include <memory>
+#include <vector>
+
+#include "core/task/include/task.hpp"
+#include "seq/gromov_a_fox_algorithm/include/ops_seq.hpp"
+
```
Comment (aobolensk) [inline tasks/seq/gromov_a_fox_algorithm/func_tests/main.cpp:12]: Please, add random tests as well

**Review by aobolensk — CHANGES_REQUESTED** [review]

**Review by aobolensk — APPROVED** [review]

File: tasks/seq/gromov_a_fox_algorithm/func_tests/main.cpp [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+  test_task_sequential.PreProcessing();
+  test_task_sequential.Run();
+  test_task_sequential.PostProcessing();
+
+  for (size_t i = 0; i < out.size(); ++i) {
+    EXPECT_NEAR(out[i], expected[i], 1e-9);
+  }
+}
+
+TEST(gromov_a_fox_algorithm_seq, test_50x50) {
```
Comment (allnes) [inline tasks/seq/gromov_a_fox_algorithm/func_tests/main.cpp:137]: please check case I x A = A without naive mult method

**Review by allnes — APPROVED** [review]
